### ISSUE TITLE
Decoding tubes

### ISSUE THREAD

**Issue by nthhiep** [MCG-NJU/MOC-Detector#16]
Hi, there's an inconsistency in the way you convert from the output of model into tubes in src/detector/decode.py/moc_decode function.
- lines 58,59: indexes of tube centers at key frame (K//2) are not modified by "mov" data
- lines 80-83: indexes of tube centers at key frame (K//2) are adjusted by adding "mov" data.

Can you explain this difference please. Furthermore, according to the lines 56,57, we have:
           xs + mov[..., 2 * i:2 * i + 1] = xs_all[:,:,i:i+1]    and 
           ys + mov[..., 2 * i + 1:2 * i + 2] = ys_all[:,:,i:i+1]  for every frame i except for i=K//2
where xs_all ys_all are tensors before beeing converted to long type (lines 61,62).

Why do you have to do this calculation again in line 80? 
Thank you in advance for the information.

**Comment by ArchiZX** [comment]
Oh, in my opinion the center frame doesn't need movement estimation. But we need to recover the numerical difference from discretization. (in ~/src/datasets/sample/sampler.py  Line#L117)

So the movement contains a local offset for each frame, and for key frame (i=K//2), movement only represents the local offset.

On the other hand, wh prediction should be gathered from current frame's action center. And the range of numerical difference is (0, 1). So indexes of tube centers at key frame (K//2) are not modified by "mov" data. (Line 58-59)

In short, xs_all and ys_all in lines 58-59 are used to gather wh prediction, but codes in lines 80-83 are proposed to count predict boxes.
